File: src/main/java/org/example/Application.java
package org.example;

import org.example.demo.GreetingsRegistrar;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Import;

@SpringBootApplication
//@Import(GreetingsRegistrarJava.class)
@Import(GreetingsRegistrar.class)
public class Application {
    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}

File: src/main/kotlin/org/example/demo/Application.kt
package org.example.demo

import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

@SpringBootApplication
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}

@Configuration
@Import(DemoBeanRegistrar::class)
class DemoConfiguration

